### PR TITLE
Fix: Editing a team member creates multiple identical cards #5394

### PR DESCRIPTION
**Pull request by leonardohofling**
Resolve https://github.com/opencollective/opencollective/issues/5394

### CODE REVIEW

File: server/graphql/v2/mutation/MemberMutations.js [hunk 1]
```diff
@@ -145,22 +145,24 @@ const memberMutations = {
...
-      // Edit member
-      const editableAttributes = pick(args, ['role', 'description', 'since']);
-
-      const [, members] = await models.Member.update(editableAttributes, {
-        returning: true,
+      const member = await models.Member.findOne({
         where: {
           MemberCollectiveId: memberAccount.id,
           CollectiveId: account.id,
+          role: [MemberRoles.ACCOUNTANT, MemberRoles.ADMIN, MemberRoles.MEMBER],
```
Comment (Betree) [inline server/graphql/v2/mutation/MemberMutations.js:152]: Can we keep only this line? It seems that the other changes (`findOne` + `update`) are not necessary

Comment (leonardohofling) [reply server/graphql/v2/mutation/MemberMutations.js:152]: @Betree yes, tested here and it worked.

Comment (leonardohofling) [reply server/graphql/v2/mutation/MemberMutations.js:156]: Updated

**Review by Betree — APPROVED** [review]

**Comment by Betree** [comment]
Perfect, thanks @leonardohofling.

There are 46 entries that need to be migrated in prod DB, I'll do that manually:

```sql
SELECT "MemberCollectiveId", "CollectiveId", "TierId", COUNT(m.id), min(m."since"), array_agg("role")
FROM "Members" m 
WHERE m."deletedAt" IS NULL
AND m.role IN ('ADMIN', 'MEMBER', 'ACCOUNTANT')
GROUP BY "MemberCollectiveId", "CollectiveId", "TierId"
HAVING count(m.id) > 1
ORDER BY max("createdAt") DESC
```

**Comment by Betree** [comment]
> Perfect, thanks @leonardohofling.
> 
> There are 46 entries that need to be migrated in prod DB, I'll do that manually:
> 
> ```sql
> SELECT "MemberCollectiveId", "CollectiveId", "TierId", COUNT(m.id), min(m."since"), array_agg("role")
> FROM "Members" m 
> WHERE m."deletedAt" IS NULL
> AND m.role IN ('ADMIN', 'MEMBER', 'ACCOUNTANT')
> GROUP BY "MemberCollectiveId", "CollectiveId", "TierId"
> HAVING count(m.id) > 1
> ORDER BY max("createdAt") DESC
> ```

A bit late, but I deleted all the impacted entries. Logs below.

<details>

```
> WITH duplicates AS (
                         SELECT "MemberCollectiveId", "CollectiveId", "TierId", COUNT(m.id), min(m."since"), array_agg("role")
                           FROM "Members" m
                           WHERE m."deletedAt" IS NULL
                           AND m.role IN ('ADMIN', 'MEMBER', 'ACCOUNTANT')
                           GROUP BY "MemberCollectiveId", "CollectiveId", "TierId"
                           HAVING count(m.id) > 1
                           ORDER BY max("createdAt") DESC
                       ) SELECT m.id, m."MemberCollectiveId", m."CollectiveId", m.role, m.since, m."createdAt", m."deletedAt"
                       FROM "Members" m
                       INNER JOIN duplicates d ON d."MemberCollectiveId" = m."MemberCollectiveId" AND d."CollectiveId" = m."CollectiveId"
                       WHERE m."deletedAt" IS NULL
                       AND m.role IN ('ADMIN', 'MEMBER', 'ACCOUNTANT')
                       ORDER BY m."MemberCollectiveId", m."CollectiveId", m.since, m.role
[2022-06-01 08:58:03] 106 rows retrieved starting from 1 in 5 s 607 ms (execution: 5 s 563 ms, fetching: 44 ms)
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 110566
[2022-06-01 09:07:21] 1 row affected in 237 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 28587
[2022-06-01 09:07:21] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 230572
[2022-06-01 09:07:21] 1 row affected in 122 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 239422
[2022-06-01 09:07:21] 1 row affected in 107 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 14653
[2022-06-01 09:07:21] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 162732
[2022-06-01 09:07:21] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 163440
[2022-06-01 09:07:21] 1 row affected in 107 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 27715
[2022-06-01 09:07:21] 1 row affected in 103 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 74166
[2022-06-01 09:07:22] 1 row affected in 103 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 298876
[2022-06-01 09:07:22] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 228918
[2022-06-01 09:07:22] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 201706
[2022-06-01 09:07:22] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 136781
[2022-06-01 09:07:22] 1 row affected in 132 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 177985
[2022-06-01 09:07:22] 1 row affected in 103 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 131426
[2022-06-01 09:07:22] 1 row affected in 103 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 41726
[2022-06-01 09:07:22] 1 row affected in 103 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 251729
[2022-06-01 09:07:22] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 240843
[2022-06-01 09:07:23] 1 row affected in 169 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 265200
[2022-06-01 09:07:23] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 265001
[2022-06-01 09:07:23] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 279675
[2022-06-01 09:07:23] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 66698
[2022-06-01 09:07:23] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 206273
[2022-06-01 09:07:23] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 63107
[2022-06-01 09:07:23] 1 row affected in 107 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 42736
[2022-06-01 09:07:23] 1 row affected in 132 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 17213
[2022-06-01 09:07:24] 1 row affected in 102 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 197252
[2022-06-01 09:07:24] 1 row affected in 109 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 213743
[2022-06-01 09:07:24] 1 row affected in 114 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 174343
[2022-06-01 09:07:24] 1 row affected in 102 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 17490
[2022-06-01 09:07:24] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 177987
[2022-06-01 09:07:24] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 170878
[2022-06-01 09:07:24] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 252631
[2022-06-01 09:07:24] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 226724
[2022-06-01 09:07:24] 1 row affected in 193 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 12713
[2022-06-01 09:07:25] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 196926
[2022-06-01 09:07:25] 1 row affected in 107 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 24565
[2022-06-01 09:07:25] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 297442
[2022-06-01 09:07:25] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 220261
[2022-06-01 09:07:25] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 13080
[2022-06-01 09:07:25] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 199622
[2022-06-01 09:07:25] 1 row affected in 104 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 255140
[2022-06-01 09:07:25] 1 row affected in 108 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 15946
[2022-06-01 09:07:25] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 13014
[2022-06-01 09:07:26] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 265452
[2022-06-01 09:07:26] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 223198
[2022-06-01 09:07:26] 1 row affected in 136 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 37251
[2022-06-01 09:07:26] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 184451
[2022-06-01 09:07:26] 1 row affected in 255 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 174327
[2022-06-01 09:07:26] 1 row affected in 107 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 248856
[2022-06-01 09:07:26] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 203717
[2022-06-01 09:07:27] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 209575
[2022-06-01 09:07:27] 1 row affected in 105 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 93048
[2022-06-01 09:07:27] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 12819
[2022-06-01 09:07:27] 1 row affected in 107 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 136773
[2022-06-01 09:07:27] 1 row affected in 106 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 273007
[2022-06-01 09:07:27] 1 row affected in 111 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 275018
[2022-06-01 09:07:27] 1 row affected in 212 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 127599
[2022-06-01 09:07:28] 1 row affected in 241 ms
> UPDATE public."Members" SET "deletedAt" = '2022-06-01 08:58:16.722000 +00:00' WHERE id = 203160
[2022-06-01 09:07:28] 1 row affected in 241 ms
```

</details>

**Pull request merged**